File: src-tauri/src/dev.rs
use crate::error::AppError;
use std::fs;
use std::path::PathBuf;
use tauri::{AppHandle, Manager};

pub fn ensure_dev_deadlock_path(app: &AppHandle) -> Result<PathBuf, AppError> {
    let app_data = app
        .path()
        .app_data_dir()
        .map_err(|e| AppError::Settings(e.to_string()))?;

    if !app_data.exists() {
        fs::create_dir_all(&app_data)?;
    }

    let dev_root = app_data.join("dev-deadlock");
    let citadel_dir = dev_root.join("game/citadel");
    let disabled_dir = citadel_dir.join("addons/.disabled");
    fs::create_dir_all(&disabled_dir)?;

    let gameinfo_path = citadel_dir.join("gameinfo.gi");
    if !gameinfo_path.exists() {
        fs::write(&gameinfo_path, "")?;
    }

    Ok(dev_root)
}

File: electron/main/services/dev.ts
import { existsSync, mkdirSync, writeFileSync } from 'fs';
import { join } from 'path';
import { getDevDeadlockPath } from '../utils/paths';

/**
 * Create or reuse a dev-mode Deadlock path for testing
 */
export function ensureDevDeadlockPath(): string {
    const devRoot = getDevDeadlockPath();
    const citadelDir = join(devRoot, 'game', 'citadel');
    const addonsDir = join(citadelDir, 'addons');
    const disabledDir = join(addonsDir, '.disabled');

    // Create directory structure
    if (!existsSync(disabledDir)) {
        mkdirSync(disabledDir, { recursive: true });
    }

    // Create empty gameinfo.gi if it doesn't exist
    const gameinfoPath = join(citadelDir, 'gameinfo.gi');
    if (!existsSync(gameinfoPath)) {
        writeFileSync(gameinfoPath, '', 'utf-8');
    }

    return devRoot;
}
